src/io/flutter/vmService/VmServiceWrapper.java

```diff
@@ -343,6 +345,52 @@ else if (errorResponses.size() > 0) {
         private void checkDone() {
           if (counter.decrementAndGet() == 0 && onFinished != null) {
             onFinished.run();
+
+            myVmService.getIsolate(isolateId, new GetIsolateConsumer() {
+              @Override
+              public void received(Isolate response) {
+                final Set<String> libraryUris = new HashSet<>();
+                for (LibraryRef library : response.getLibraries()) {
+                  libraryUris.add(library.getUri());
+                }
+
+                final ElementList<Breakpoint> breakpoints = response.getBreakpoints();
+                if (breakpoints.isEmpty()) {
+                  return;
+                }
+
+                final Analytics analytics = FlutterInitializer.getAnalytics();
+                final String category = "breakpoint";
+                boolean scriptFound = false;
+                for (Breakpoint breakpoint : breakpoints) {
+                  Object location = breakpoint.getLocation();
+                  if (location instanceof UnresolvedSourceLocation) {
+                    final ScriptRef script = ((UnresolvedSourceLocation)location).getScript();
+                    if (script != null) {
+                      if (libraryUris.contains(script.getUri())) {
+                        // Record to analytics that at least one breakpoint has been mapped to a library.
+                        analytics.sendEvent(category, "mapping-found");
```

Comment:
i'd rather track the count of the # of resolved and unresolved breakpoints than short circuit as soon as we found one successful breakpoint. Naively, something has gone wrong if any of the breakpoints are unresolved and do not match a source locations from the loaded libraries. Perhaps I'm missing something and there is a risk of false positives?

Reply:
Each set breakpoint generates multiple `Breakpoint` instances that show up here, so I'm not sure how to capture the total number of breakpoints from the user's viewpoint. e.g. for an external project's breakpoint, I see 2 `Breakpoint` instances - one's location has a `script` field with a URI that is associated with a package (this one can be considered "mapped"), the other has a `scriptUri` field instead that appears to be the location of a file on the device.

For a bazel project with a successful breakpoint, there are 3 `Breakpoint` instances - two are similar as the external project, plus one that has a `scriptUri` field that is the non-package location of the file.

So I'm not sure, maybe there's a better way to track a user view of how many breakpoints have been set earlier on in this process?